app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -11,6 +11,8 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
       default_hostname: '',
       amqp_hostname: '',
       hawkular_hostname: '',
+      provider_options_values: {},
```

Comment:
I kind of need the same object to hold both the descriptions of the options and the values. This way it is much easier to iterate over the descriptions in the haml file and find the object that should hold the value.
Another option can be to hold it outside of emsCommonModel and only copy the value of the options back to it when the user clicks "save". WDYT?